### DISCUSSION TITLE
[Authenitcation Service] Can I use the groups of my Identity Provider?

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#327] [Q&A] [answered]
### Which section?

Marketplace

### Which is the version?

-

### Description

Hi!

I have an external Identity Provider that already comes with the concept of groups. I would like to use the Authentication Service plugin but, as far as I understand from documentation, groups are assigned to users using a dedicated CRUD collection. Is there a way to use the plugin and manage the groups directy on my Identity Provider?

Thanks!

**Comment by davidebianchi** [comment, marked as answer]
Hi @Lucascanna! No, it is not possible to inherit groups from idp at the moment. The only keys which inherit from idp are name, email and id (and only at login time). 
When the user login, the JWT is generated from those data.

**Reply by lucascanna** [reply]
Thanks @davidebianchi. So, the JWT is generated at login time (route `/oauth/token`) by merging name, email and id from the IDP and the other properties from CRUD collection. Then, the `/userinfo` route only verifies and returns the JWT payload, without contacting again neither the IDP nor the CRUD. Right?

**Reply by davidebianchi** [reply]
Yes correct, it is the actual flow.
